lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
+ * @returns {Object} Analytics payload
+ */
+function fetchPlatformData(context, eventName) {
     const os = require('os');
-    const request = require('request');
     const { machineId } = require("node-machine-id");
+    const environment = utils.getContext();
+
+    return new Promise((resolve, reject) => {
+        machineId()
+        .catch((e) => {
+            reject('Analytics submission failed');
+            return;
+        })
+
+        .then((identifier) => {
+            if (!identifier) {
+                identifier = sha256('unknown');
+            }
 
-    module.exports = function(eventName, context) {
-        // Submit analytics after some time - we do this to give the application a chance to disable analytics.
+            // payloads for webhook and MixPanel differ slightly
+            const payload = {
+                'webHook': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': 'javascript',
+                        'Target': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                },
+                'mixPanel': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                }
+            };
+
+            resolve(payload);
+            return;
+        });
+    });
+}
+
+/**
+ * Generate and submit analytics data to servers.
+ * @param  {[webhookProdDetails|webhookStageDetails|mixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Promise} Promise that resolves when analytics data has been submitted
+ */
+async function submitAnalytics(requestFactory, context, eventName) {
```

Comment:
I think this function could be greatly simplified if it was simply awaiting a promise which resolved after 100ms.

```
await new Promise(resolve => setTimeout(resolve, 100));
```

If you did that, then you can `await` the result of the call to `fetchPlatformData` and avoid using the `.then` and `.catch` methods, which are the "older" and more complicated way to program with promises.